### ISSUE TITLE
Unable to override a getter when inheriting from a mixin class

### ISSUE THREAD

**Issue by whzx5byb** [microsoft/TypeScript#44938]
# Bug Report

### 🔎 Search Terms

mixin getter override

### 🕗 Version & Regression Information

- This is the behavior in every version I tried

### ⏯ Playground Link

[Playground link with relevant code](https://www.typescriptlang.org/play?#code/MYGwhgzhAECC0G8CwAoa7rAPYDsIBcAnAV2Hy0IAoA6WsQgcwgC5owcBPAbQF0BKRAF9UGaAwCm+aAFsOAOTDTxlPqwKEAljgaJoI0RkKTihHNADkscwG49aDMJSPUAM2I4yG3NACyGgB5aADwAKtDi-vjiOAAmMPgcAA7iWC5wAHyUAMrEyYSsIQLI9uhG+CZmoJAwAELhkdFx0Dl5iPoGYpIy8orKqtDqWjrFHR1lFRY1Nu0GjqKOzihVUNAAwvVRsTB+gTiUsEUzElKyCkoqakRDujOi46YWq+Z286iLAG70mNAAvNA44gA7msVNZUNg8FgQOJqCAsAxKMBqKdenxbAB6dHQQEaEAgaCJTQ4KTmJ5sGARZJkcQxaD4AAWWGIDHpQA)

### 💻 Code

```ts
class A {
    constructor(...args: any[]) {}
    get myName(): string { 
        return 'A'; 
    }
}

function Mixin<T extends typeof A>(Super: T) {
    return class B extends Super {
        get myName(): string {
            return 'B';
        }
    }
}

class C extends Mixin(A) {
    get myName(): string { 
    // Unexpected Error:
    // 'myName' is defined as a property in class 'Mixin<typeof A>.B & A', but is overridden here in 'C' as an accessor.
        return 'C' 
    }
}

var c = new C();
console.log(c.myName); // will print 'C' as expected though

```

### 🙁 Actual behavior

Error in `get myName()` of class C:
`'myName' is defined as a property in class 'Mixin<typeof A>.B & A', but is overridden here in 'C' as an accessor.` 

It is obviously incorrect because `myName` is defined as a getter (NOT a property) in either A or B.

### 🙂 Expected behavior

No error.
